docs/opc-event-output.md

```diff
@@ -0,0 +1,199 @@
+# Detailing the output for OPC UA Events in OPC Publisher
+
+This section describes what the output looks like when listening for events in the OPC Publisher. Events are produced in the PubSub format specified in the OPC UA Standard. The payload is an event which consists of fields selected in the select clause and its values.
+
+Here is an example of the output we get when listening to events from the Simple Events sample:
+```json
+{
+  "body": [
+    {
+      "MessageId": "43",
+      "MessageType": "ua-data",
+      "PublisherId": "SIMPLE-EVENTS",
+      "DataSetWriterGroup": "SIMPLE-EVENTS",
+      "Messages": [
+        {
+          "DataSetWriterId": "SIMPLE-EVENTS",
+          "MetaDataVersion": {
+            "MajorVersion": 1,
+            "MinorVersion": 0
+          },
+          "Payload": {
+            "SimpleEvents": {
+              "Value": {
+                "EventId": "+6CQjN1eqkO6+yHJnxMz5w==",
+                "EventType": "http://microsoft.com/Opc/OpcPlc/SimpleEvents#i=14",
+                "Message": "The system cycle '59' has started.",
+                "ReceiveTime": "2021-06-21T12:38:55.5814091Z",
+                "Severity": 1,
+                "SourceName": "System",
+                "SourceNode": "i=2253",
+                "Time": "2021-06-21T12:38:55.5814078Z"
+              },
+              "SourceTimestamp": "2021-06-21T12:38:55.5814091Z"
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "enqueuedTime": "Mon Jun 21 2021 14:39:02 GMT+0200 (Central European Summer Time)",
+  "properties": {
+    "$$ContentType": "application/x-network-message-json-v1",
+    "iothub-message-schema": "application/ua+json",
+    "$$ContentEncoding": "utf-8"
+  }
+}
+```
+A few things to note here. Here the event in the payload is named "SimpleEvents" which has been configured to by setting the DisplayName property of the events configuration. Also note that all fields+values reside under the Value key.
+
+The format produced here does not contain enough information to decode the JSON properly. If you need this there is a command-line switch in the OPC Publisher called "reversibleencoding", or "re" which can be set to either true or false. The default value is false. If you enable this setting the output will look like this:
+```json
+{
+  "body": [
+    {
+      "MessageId": "15",
+      "MessageType": "ua-data",
+      "PublisherId": "SIMPLE-EVENTS",
+      "DataSetWriterGroup": "SIMPLE-EVENTS",
+      "Messages": [
+        {
+          "DataSetWriterId": "SIMPLE-EVENTS",
+          "MetaDataVersion": {
+            "MajorVersion": 1,
+            "MinorVersion": 0
+          },
+          "Payload": {
+            "SimpleEvents": {
+              "Value": {
+                "Type": "ExtensionObject",
+                "Body": {
+                  "TypeId": "s=EncodeableDictionary",
+                  "Encoding": "Json",
+                  "Body": {
+                    "_Count": 8,
```

Comment:
Please update to latest changes, _Count and _Keys have to be removed

Reply:
Done!